rust/main/hyperlane-base/src/types/gcs_storage.rs

```diff
@@ -195,28 +216,42 @@ impl CheckpointSyncer for GcsStorageClient {
 
     /// Write the agent metadata to this syncer
     async fn write_metadata(&self, metadata: &AgentMetadata) -> Result<()> {
+        let object_name = self.object_path(METADATA_KEY);
         let serialized_metadata = serde_json::to_string_pretty(metadata)?;
-        self.inner
-            .insert_object(&self.bucket, METADATA_KEY, serialized_metadata)
-            .await?;
-        Ok(())
+
+        match self.inner.insert_object(&self.bucket, &object_name, serialized_metadata.into_bytes()).await {
+            Ok(_) => {
+                info!("Successfully uploaded metadata to '{}'", object_name);
+                Ok(())
+            }
+            Err(e) => {
+                error!("Failed to upload metadata to '{}': {:?}", object_name, e);
+                Err(e.into())
+            }
+        }
```

Comment:
since there are 5 calls to `insert_object` in this file, it'd make sense to refactor this into a function that just logs `Successfully posted to GCS storage` / `Failed to [...]` without going into specifics, and then `#[instrument(skip(...)]`-ing the functions that call this so that they show up in the log `spans` (while skipping all function arguments to avoid polluting the logs). So basically in `write_latest_index`, `write_checkpoint`, `write_metadata`, `write_announcement`, `write_reorg_status`

Reply:
Fixed the above edits:
https://github.com/IvanPsurtcev/hyperlane-monorepo/commit/9eec9548afb406428016571fc383de593e42da8c